google_gke/locals.tf

```diff
@@ -1,69 +1,49 @@
-    istio                      = false
-    vertical_pod_autoscaling   = false
-    cloudrun                   = false
+  cluster_name        = "${var.name}-${var.realm}"
+  cluster_network_tag = "gke-${local.cluster_name}"
+
+  labels_defaults = {
+    "realm"     = var.realm
+    "name"      = var.name
+    "region"    = var.region
+    "terraform" = "true"
   }
-  cluster_addons = merge(local.cluster_addons_defaults, var.cluster_addons)
-
-  cluster_features_defaults = {
-    "velero"             = true
-    "external_secrets"   = true
-    "prometheus"         = false
-    "flux"               = false
-    "flux_helm_operator" = false
-  }
-  cluster_features = merge(local.cluster_features_defaults, var.cluster_features)
-
-  velero_defaults = {
-    "credentials.useSecret"                                                = false
-    "configuration.provider"                                               = "gcp"
-    "configuration.backupStorageLocation.name"                             = "gcp"
-    "configuration.backupStorageLocation.bucket"                           = local.cluster_features["velero"] ? google_storage_bucket.bucket[0].name : ""
-    "configuration.backupStorageLocation.config.serviceAccount"            = module.velero_workload_identity.gcp_service_account_email
-    "configuration.volumeSnapshotLocation.name"                            = "gcp"
-    "initContainers[0].name"                                               = "velero-plugin-for-gcp"
-    "initContainers[0].image"                                              = "velero/velero-plugin-for-gcp:v1.1.0"
-    "initContainers[0].volumeMounts[0].mountPath"                          = "/target"
-    "initContainers[0].volumeMounts[0].name"                               = "plugins"
-    "serviceAccount.server.name"                                           = "velero"
-    "serviceAccount.server.annotations.iam\\.gke\\.io/gcp-service-account" = module.velero_workload_identity.gcp_service_account_email
-    "schedules.daily.schedule"                                             = "0 0 * * *"
-    "schedules.daily.template.ttl"                                         = "720h0m0s"
-    "schedules.daily.template.storageLocation"                             = "gcp"
-    "schedules.daily.template.volumeSnapshotLocations[0]"                  = "gcp"
-    "schedules.daily.template.includedNamespaces[0]"                       = "*"
+  labels     = merge(local.labels_defaults, var.labels)
+  project_id = trim(data.google_project.project.id, "/projects/")
+
+  tags_defaults = [var.realm, var.name, var.region, "terraform"]
+  tags          = setunion(local.tags_defaults, var.tags)
+
+  # internal networking setup
+  datapath_provider = var.enable_dataplane ? "ADVANCED_DATAPATH" : "DATAPATH_PROVIDER_UNSPECIFIED"
+
+  # monitoring setup
+  resource_usage_export_dataset_id = var.create_resource_usage_export_dataset ? google_bigquery_dataset.dataset[0].id : var.resource_usage_export_dataset_id
```

Comment:
instead of `google_bigquery_dataset.dataset[0].id` it seems like this needs to be `google_bigquery_dataset.dataset[0].dataset_id`

From my testing, `id` gives the whole path(`projects/{{project}}/datasets/{{dataset_id}}`) but the module seems to only want the `dataset_id` and won't accept the whole path. Confirmed my testing works with the suggestion above.

Reply:
changed